src/resources/views/character/mail-timeline.blade.php

```diff
@@ -77,7 +76,30 @@
             @endforeach
 
           </li>
-          @endif
+
+        @endif
+
+        @if($message->recipients->where('recipient_type', 'mailing_list')->count() > 0)
+
+          <li>
+            <b>{{ trans('web::seat.to_mailing_list') }}:</b>
+
+            @foreach($message->recipients->where('recipient_type', 'mailing_list') as $recipient)
+
+              @if (! is_null($recipient->mailing_list))
+                {{ $recipient->mailing_list->name }}
+              @else
+                {{ trans('web::seat.unknown') }}
+              @endif
+
+            @endforeach
+
+          </li>
+
+        @endif
+
+        <b>{{ trans('web::seat.source') }}: </b>
+        @include('web::partials.character', ['character' => $CharacterInfo::find($message->character_id) ?: $message->character_id, 'character_id' => $message->character_id])
```

Comment:
Sorry, but model look ups in views are not a type of complexity I think we should be adding.

Reply:
overall i agree, however the whole view and controller for mail-timeline needs to have a proper refactoring. This model look ups are already across this view and would not add to its complexity. It is however a quick application to add value to the view without putting in to much effort. 

Having said that. I am aware that this view needs a redo as the way it is build up is way to complex.